### ISSUE TITLE
Icon detection is borked

### ISSUE THREAD

**Issue by HitomiTenshi** [file-icons/atom#511]
The latest update (**2.0.8**) has messed up the icon detection.

![2017-01-11_10-00-13](https://cloud.githubusercontent.com/assets/7304212/21842164/d5926b40-d7e5-11e6-8c39-6ad4b2f4c33a.gif)

![atom_2017-01-11_10-02-51](https://cloud.githubusercontent.com/assets/7304212/21842170/d9c16464-d7e5-11e6-9a3f-40bb578fbb6d.png)

Everything worked fine with **2.0.7**.
I'm using Atom 13.0.

**Comment by Alhadis** [comment]
Can you tell us what packages you're using?

**Comment by HitomiTenshi** [comment]
Community Packages:
- atom-typescript 10.1.13
- autoclose-html 0.23.0
- color-picker 2.2.3
- editorconfig 2.1.0
- file-icons 2.0.8
- highlight-selected 0.11.2
- linter 1.11.21
- merge-conflicts 1.4.4
- minimap 4.26.5
- minimap-git-diff 4.3.1
- minimap-highlight-selected 4.4.0
- minimap-pigments 0.2.2
- minimap-selection 4.4.0
- pigments 0.38.0

I also have a few core packages **disabled**:
- about
- background-tips
- exception-reporting
- go-to-line
- metrics
- spell-check
- welcome

**Comment by Alhadis** [comment]
What CSS classes are being applied to those icons when they change? Is there anything insightful in the dev console?

Most importantly, does this persist between restarts? I take it from your cursor that you're running Windows 10?

**Comment by HitomiTenshi** [comment]
Yes I'm using Windows 10 Pro x64.

It persists between restarts.

Looking at **package.json's** classes (See the .gif in the first post):

Clicking on it gets: `name icon npm-icon medium-red`
Clicking on systemjs.config.js sets it to: `name icon js-icon medium-yellow`
Clicking on tsconfig.aot.json sets it to: `name icon database-icon medium-yellow`

**Comment by Alhadis** [comment]
Okay, first of all, File-Icons actually doesn't change anything in response to click-events. So it sounds like another package is at fault here. Can you disable every other package *except* file-icons and tell me if this still happens?

If not, try re-enabling them each one-by-one until you find the one that's causing this oddity.

As for the ColourBuffer error, I can only fathom it's related to the `color-picker` or `pigments` packages, because nothing of that name exists in File-Icons...

**Comment by HitomiTenshi** [comment]
I disabled every package except file-icons and it still happens 😕

**Comment by Alhadis** [comment]
And restarted Atom?

**Comment by HitomiTenshi** [comment]
and restarted atom yes

**Comment by Alhadis** [comment]
That's it, I have to develop a diagnostics command to generate some sort of proper report from the workspace state, because these blind interrogations are getting ridiculous... **sigh**

**Comment by HitomiTenshi** [comment]
Alright so I re-enabled all core packages and it still happens.

![2017-01-11_10-39-23](https://cloud.githubusercontent.com/assets/7304212/21843198/61aaaba2-d7ea-11e6-8cfb-1a32037ae3e1.gif)

**Comment by Alhadis** [comment]
Okay, last resort questions, which are really just stabs in the dark:

1. Have you tried [clearing the cache?](https://github.com/file-icons/atom#troubleshooting) I'm guessing that's a given...

2. Is this happening with ONLY these particular files? Are other projects unaffected?

3. Does disabling certain icon-matching strategies help? (In the package settings)

If none of the above help, exorcise or burn your computer, and buy one that isn't possessed by a demonic spirit.

**Comment by HitomiTenshi** [comment]
1. Clearing cache and restarting doesn't help.
2. Well there are more files affected, I just clicked through all of them. See this gif:

![2017-01-11_10-52-13](https://cloud.githubusercontent.com/assets/7304212/21843648/0e28aae0-d7ec-11e6-9a60-387541b84360.gif)

3. Weird... disabling all matching strategies fixed the 3 files that were changing on click, they now stay as a database icon. But the .gitignore still changes between the git and the sass icons.

My guess is that file-icons has a problem with the folder called "sass".

Ok I renamed it to "test", cleared the cache, restarted and it's still doing the spooky stuff. On a small side note though, even after renaming the folder to test, clearing cache and restarting, it still has the js icon.

**Comment by Alhadis** [comment]
Wait, I forgot to ask about themes... what do you have installed? Does switching them help?

Does your user config define any custom filetypes?

Jesus, this is really a shitfight.

**Comment by HitomiTenshi** [comment]
Holy Jesus!

I have the default ui theme "One Dark" and the default syntax theme "One Dark".

I just switched between themes, cleared the cache and restarted atom (all of that multiple times and even with other projects open). It didn't help.

By the way, I just noticed this bug only happens with that particular project. Some detection mechanism seems to be bugged with that project for some reason.

My other projects work fine and show all the right icons.

**Comment by Alhadis** [comment]
Congratulations. You've managed to download a poltergeist. How does it feel?

If you're not shitting yourself yet, you should be.

(Honestly, I'm as puzzled as you are. I've got nothing... this shouldn't be happening, especially not on a per-project basis... :\

**Comment by HitomiTenshi** [comment]
Alright, this is the weirdest thing ever. I just committed and pushed the latest changes and after the assets folder turned from green to gray it stopped doing the spooky stuff. Everything suddenly became normal again. Right on the moment when I clicked on atom after committing and pushing the changes... 👻 

I tried changing stuff in the assets folder and creating a new folder called just "asset" but the bug is gone now. Maybe it was an issue with a git related package?

Anyhow, this is a case for the ghost busters. Sorry for taking up your time with this spooky shit. If I should get this one again I'll just commit and push.
